feat(dialog): add new `keyboardEnabled` prop (#13178)
youzan/vant: packages/vant/src/dialog/README.md

| beforeClose | Callback function before close | _(action: string) => boolean \| Promise\<boolean\>_ | - |
| transition | Transition, equivalent to `name` prop of [transition](https://vuejs.org/api/built-in-components.html#transition) | _string_ | - |
| teleport | Specifies a target element where Dialog will be mounted | _string \| Element_ | `body` |

> | keyboardEnabled | Whether or not the keyboard capability is used, the `Enter` and `Esc` keys by default execute the confirm and cancel functions when the `confirm` and `cancel` buttons are displayed | _boolean_ | `true` |

Reviewer: ```suggestion
| keyboardEnabled | Whether to enable keyboard capabilities. When displaying the confirm and cancel buttons, the keyboard's `Enter` and `Esc` will call the `confirm` and `cancel` functions by default | _boolean_ | `true` |
```

| before-close | Callback function before close | _(action: string) => boolean \| Promise\<boolean\>_ | - |
| transition | Transition, equivalent to `name` prop of [transition](https://vuejs.org/api/built-in-components.html#transition) | _string_ | - |
| teleport | Specifies a target element where Dialog will be mounted | _string \| Element_ | - |

> | keyboard-enabled | Whether or not the keyboard capability is used, the `Enter` and `Esc` keys by default execute the confirm and cancel functions when the `confirm` and `cancel` buttons are displayed | _boolean_ | `true` |

Reviewer: ```suggestion
| keyboard-enabled | Whether to enable keyboard capabilities. When displaying the confirm and cancel buttons, the keyboard's `Enter` and `Esc` will call the `confirm` and `cancel` functions by default | _boolean_ | `true` |
```
